File: GameStats.Common/Contracts/Player/PlayerDto.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace GameStats.Common.Contracts.Player
{
    public class PlayerDto
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }

        [JsonPropertyName("lastName")]
        public string LastName { get; set; }

        [JsonPropertyName("jerseyNumber")]
        public int JerseyNumber { get; set; }

        [JsonPropertyName("teamId")]
        public int TeamId { get; set; }

        [JsonPropertyName("teamName")]
        public string TeamName { get; set; }

        [JsonPropertyName("dateOfBirth")]
        public DateTime DateOfBirth { get; set; }

        [JsonPropertyName("gender")]
        public string Gender { get; set; }
    }

    public class PlayerListDto
    {
        public List<PlayerDto> Players { get; set; } = new();
    }
}

File: GameStats.React/ClientApp/src/DTOs/Player/PlayerDto.js
export default class PlayerDto {
    constructor(options = {}) {
        this.id = options.id || 0;
        this.firstName = options.firstName || '';
        this.lastName = options.lastName || '';
        this.jerseyNumber = options.jerseyNumber;
        this.teamId = options.teamId || 0;
        this.teamName = options.teamName || '';
        this.dateOfBirth = options.dateOfBirth || '';
        this.gender = options.gender || '';

        this.seasonId = options.seasonId || 0;
        this.seasonDescription = options.seasonDescription || '';
        this.leagueId = options.leagueId || 0;
        this.leagueDescription = options.leagueDescription || '';

        this.name = `${this.firstName} ${this.lastName}`;
    }

    stringify() {
        return JSON.stringify(this);
    }
}